pennylane/io/qualtran_io.py

```diff
@@ -48,386 +46,28 @@
-    return gate_types
-
-
-@_get_op_call_graph.register
-def _(op: qtemps.subroutines.QSVT):
-    """Call graph for Quantum Singular Value Transform"""
-
-    # From ResouceQSVT
-    gate_types = defaultdict(int, {})
-    UA = op.hyperparameters["UA"]
-    projectors = op.hyperparameters["projectors"]
-    num_projectors = len(projectors)
-
-    for proj_op in projectors[:-1]:
-        gate_types[_map_to_bloq(proj_op)] += 1
-
-    gate_types[_map_to_bloq(UA)] += num_projectors // 2
-    gate_types[_map_to_bloq(UA).adjoint()] += (num_projectors - 1) // 2
-    gate_types[_map_to_bloq(projectors[-1])] += 1
-
-    return gate_types
-
-
-@_get_op_call_graph.register
-def _(op: qtemps.subroutines.Select):
-    """Call graph for Select"""
-
-    # From ResourceSelect
-    gate_types = defaultdict(int, {})
-    ops = op.hyperparameters["ops"]
-    cmpr_ops = [_map_to_bloq(op) for op in ops]
-
-    x = qt_gates.XGate()
-
-    num_ops = len(cmpr_ops)
-    num_ctrl_wires = int(np.ceil(np.log2(num_ops)))
-    num_total_ctrl_possibilities = 2**num_ctrl_wires  # 2^n
-
-    num_zero_controls = num_total_ctrl_possibilities // 2
-    gate_types[x] = num_zero_controls * 2  # conjugate 0 controls
-
-    for cmp_rep in cmpr_ops:
-        ctrl_op = cmp_rep.controlled(CtrlSpec(cvs=[1] * num_ctrl_wires))
-        if cmp_rep == qt_gates.XGate() and num_ctrl_wires == 1:
-            ctrl_op = qt_gates.CNOT()
-        gate_types[ctrl_op] += 1
-
-    return gate_types
-
-
-@_get_op_call_graph.register
-def _(op: qops.StatePrep):
-    """Call graph for StatePrep"""
-
-    # MottonenStatePrep
-    gate_types = defaultdict(int, {})
-    num_wires = len(op.wires)
-    rz = qt_gates.Rz(0)
-    cnot = qt_gates.CNOT()
-
-    r_count = 2 ** (num_wires + 2) - 5
-    cnot_count = 2 ** (num_wires + 2) - 4 * num_wires - 4
-
-    if r_count:
-        gate_types[rz] = r_count
-
-    if cnot_count:
-        gate_types[cnot] = cnot_count
-    return gate_types
-
-
-@_get_op_call_graph.register
-def _(op: qtemps.subroutines.ModExp):
-    """Call graph for ModExp"""
-
-    # From ResourceModExp
-    mod = op.hyperparameters["mod"]
-    num_work_wires = len(op.hyperparameters["work_wires"])
-    num_x_wires = len(op.hyperparameters["x_wires"])
-
-    if mod == 2**num_x_wires:
-        num_aux_wires = num_x_wires
-        num_aux_swap = num_x_wires
-    else:
-        num_aux_wires = num_work_wires - 1
-        num_aux_swap = num_aux_wires - 1
-
-    qft = _map_to_bloq(qtemps.QFT(wires=range(num_aux_wires)), map_ops=False)
-    qft_dag = qft.adjoint()
-
-    sequence = _map_to_bloq(
-        qtemps.ControlledSequence(
-            qtemps.PhaseAdder(k=3, x_wires=range(1, num_x_wires + 1)), control=[0]
-        )
-    )
-    sequence_dag = sequence.adjoint()
-
-    cnot = qt_gates.CNOT()
-
-    mult_resources = {
-        qft: 2,
-        qft_dag: 2,
-        sequence: 1,
-        sequence_dag: 1,
-        cnot: min(num_x_wires, num_aux_swap),
+def _get_op_call_graph(op):
```

Comment:
What happens to the non-trivial mapping?

Reply:
It's handled by estimate.